cpp/src/transform/transform.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2019, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/cudf.h>
+#include <jit/type.h>
+#include <nvstrings/NVCategory.h>
+#include <utilities/cudf_utils.h>
+#include <bitmask/bitmask_ops.hpp>
+#include <bitmask/legacy_bitmask.hpp>
+#include <cudf/copying.hpp>
+#include <string/nvcategory_util.hpp>
+#include <utilities/error_utils.hpp>
+#include "jit/core/launcher.h"
+
+#include <utilities/column_utils.hpp>
+
+namespace cudf {
+namespace transformation {
+
+/**---------------------------------------------------------------------------*
+ * @brief Computes output valid mask for op between a column and a scalar
+ *
+ * @param out_null_coun[out] number of nulls in output
+ * @param valid_out preallocated output mask
+ * @param valid_col input mask of column
+ * @param num_values number of values in input mask valid_col
+ *---------------------------------------------------------------------------**/
+void col_valid_mask_and(gdf_size_type& out_null_count,
```

Comment:
Is it possible for `transform` to update the null values of the output? That is, I'd expect that the only null values in the output would be the null values in the input, and therefore you can simply copy the bitmask from the input to the output.

Reply:
Since there are several `if` conditions that need to be checked my thought is I will keep this function but update `null_count` of the output in the `transform` by simply equating that to the `null_count` of the input?

Reply:
I'm not following. At best, this function looks to be redundant with the one used in the binary ops. More importantly, I don't see why it's necessary at all given that the output's `valid` and `null_count` should just be simply copied from the input.

Reply:
Agreed. The checks are redundant after the checks that already exist in `transform`.